### ISSUE TITLE
Title block doesn't respect "Disable new blocks" restriction from DX Layout

### ISSUE THREAD

**Issue by avoinea** [plone/volto#6306]
**Describe the bug**

[Title-Layout.webm](https://github.com/user-attachments/assets/470cca6d-d14d-4f30-afe6-467bf0255498)

**To Reproduce**

1. Go to ControlPanel > Content Types
2. Add new content type: Book
3. Go to Layout for Book and enable editable blocks
4. Click on Title block and then on Settings in the Sidebar
5. Select "Disable new blocks" you will notice the "+" disappears and Save
6. Go to Home Page and add a new Book
7. Fill the Title block and press Enter. 

**Expected behavior**
A new block **should not** be created when you press Enter in the Title block if it has "Disable new blocks" 

**Screenshots**

**Software (please complete the following information):**

- OS: macOS
- Browser Chrome
- Volto Version 18.0.0-alpha.43
- Plone Version 6.0.13
- Plone REST API Version 9.7.2

**Additional context**

**Comment by MAX-786** [comment]
Hi @avoinea , i came across the same issue! you got any lead on fixing it?

**Comment by avoinea** [comment]
@MAX-786 Here is a starting point 

https://github.com/plone/volto/blob/157df05e5da7cc4349f2fefaa6e3566da56a8213/packages/volto/src/components/manage/Blocks/Block/Edit.jsx#L176-L186

**Comment by avoinea** [comment]
vs

https://github.com/plone/volto/blob/157df05e5da7cc4349f2fefaa6e3566da56a8213/packages/volto/src/components/manage/Blocks/Title/Edit.jsx#L151

**Comment by MAX-786** [comment]
thanks @avoinea  and I looked briefly and i think problem for **Title Block** lies here:

https://github.com/plone/volto/blob/157df05e5da7cc4349f2fefaa6e3566da56a8213/packages/volto/src/components/manage/Blocks/Title/Edit.jsx#L71

So, i don't have solid idea about 'detached' but i think it checks for 'blockHasOwnFocusManagement', i might be wrong but doesn't matter and the solution would be simple with jsut adding new flag like 'disableNewBlocksByDefault' and check for both flags like 
```js
  const disableNewBlocksByDefault = useMemo(() => data?.disableNewBlocks, [data?.disableNewBlocks]);
```
And check for it here:

https://github.com/plone/volto/blob/157df05e5da7cc4349f2fefaa6e3566da56a8213/packages/volto/src/components/manage/Blocks/Title/Edit.jsx#L107-L111



Now for Text Block, i think it has to do with slate because if you enter when its empty and 'disableNewBlocks' is true, it removes the block itself instead of doing 'nothing' and when you type in something and press enter it adds new block which it should not.
This might be because slate is overriding the 'data' prop of 'Block/Edit' compnent which contains 'disableNewBlocks' flag and removes it.
So the fix for this might be more than one line fix but IG we can open new ticket  targeting that issue and I can open a PR for fixing title issue separately?

**Comment by stevepiercy** [comment]
@avoinea I think that this goes beyond creating a new content type, but also affects editing existing content types. Using the demo site:

1. Login, navigate to Content Types control panel
2. Click `Page`
3. Under the `Behaviors` tab, there are two options:
   - `Blocks` / Enables Volto Blocks support
   - `Blocks (Editable Layout)` / Enables Volto Blocks (editable layout) support

   ![Image](https://github.com/user-attachments/assets/760cd7b4-a68b-4a3d-a2cd-eff1b78e283f)

   Check `Blocks (Editable Layout)` and click `Save`.
4. Click `More` > `Layout`. You will see the message "Can not edit Layout for Page content-type as the Blocks behavior is enabled and read-only" and a button `Enable editable Blocks`. This makes me think that the saved setting in Item 3 is not actually saved, or conflicts with the `Blocks` setting.

   ![Image](https://github.com/user-attachments/assets/f6a87dff-3d69-4c89-bb96-cd7c306e826f)

5. Continue and click `Enable editable Blocks`.
6. Click the `Settings` tab, click in the Title block, and check `Disable new blocks`.
7. Click the `Settings` tab, click in the Text block, and check `Disable new blocks`, and click `Save`. There is no toast indicating that the setting was saved.
8. Create a new page, edit the title, and hit `Enter`. A new Text block is created. WAT?! Undo with CMD-Z.
9. Edit the existing Text block, and hit `Enter`. The Text block disappears. WAT?!

As an aside, I have no idea what is the difference between these two settings under Item 3, and I couldn't find anything helpful in the docs https://6.docs.plone.org/search.html?q=editable+layout. I wonder if they should be paired as radio buttons or a select menu instead of checkboxes?

In summary, I think we need to first document the correct behavior, then try to address the multiple issues in this issue.
